lib/service/storage/elasticsearch.ts

```diff
@@ -3218,9 +3436,48 @@ class ElasticSearch extends Service {
       searchBody.query = { match_all: {} };
     }
 
+    if (index && this.virtualIndex.isVirtual(index)) {
+      this._sanitizeSearchBodyForVirtualIndex(searchBody, index);
+    }
+
     return searchBody;
   }
 
+  _sanitizeSearchBodyForVirtualIndex(searchBody, index) {
+    const filteredQuery = {
+      bool: {
+        filter: [],
+        must: {},
+        should: null,
+      },
+    };
+    if (searchBody.query.bool) {
+      filteredQuery.bool = searchBody.query.bool;
+      if (!filteredQuery.bool.filter) {
+        filteredQuery.bool.filter = []; //add query bool in filter
+      } else if (!Array.isArray(filteredQuery.bool.filter)) {
+        filteredQuery.bool.filter = [filteredQuery.bool.filter];
+      }
+    } else {
+      filteredQuery.bool.must = searchBody.query;
```

Comment:
I think this create a circular dependency no?

Reply:
I don't understand why.

Reply:
Right, I will correct that!